game/missiongenerator/tgogenerator.py

```
@@ -360,10 +352,11 @@ def generate(self) -> None:
             brc = self.steam_into_wind(ship_group)
 
             # Set Carrier Specific Options
-            if g_id == 0:
+            carrier_type = self.get_carrier_type()
```

as only group 0 is the carrier group there is no need to get the type for any other group here. 
Also we should raise an error if this returns None as then something else went quite wrong as we are in the carrier generator so this should never ever be None

---

I have now changed this to match your suggestions.